app/services/SessionCacheService.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import common.Dates.constructDate
+import common.KeystoreKeys.ResidentPropertyKeys
+import connectors.SessionCacheConnector
+import models.resident._
+import models.resident.properties._
+import models.resident.properties.gain.{OwnerBeforeLegislationStartModel, WhoDidYouGiveItToModel, WorthWhenGiftedModel}
+import play.api.mvc.Results.Redirect
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.frontend.exceptions.ApplicationException
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object SessionCacheService extends SessionCacheService{
+   lazy val sessionCacheConnector = SessionCacheConnector
+}
+
+trait SessionCacheService {
+  val sessionCacheConnector: SessionCacheConnector
+  lazy val homeLink = sessionCacheConnector.homeLink
```

Comment:
This doesn't need to be a val if it is already in the connector. Just use the connectors value instead of reassigning it.